x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/aggs/InferencePipelineAggregator.java

```diff
@@ -0,0 +1,132 @@
+
+        InternalMultiBucketAggregation<InternalMultiBucketAggregation, InternalMultiBucketAggregation.InternalBucket> originalAgg =
+            (InternalMultiBucketAggregation<InternalMultiBucketAggregation, InternalMultiBucketAggregation.InternalBucket>) aggregation;
+        List<? extends InternalMultiBucketAggregation.InternalBucket> buckets = originalAgg.getBuckets();
+
+        List<InternalMultiBucketAggregation.InternalBucket> newBuckets = new ArrayList<>();
+        for (InternalMultiBucketAggregation.InternalBucket bucket : buckets) {
+            Map<String, Object> inputFields = new HashMap<>();
+
+            if (bucket.getDocCount() == 0) {
+                // ignore this empty bucket unless the doc count is used
+                if (bucketPathMap.containsKey("_count") == false) {
+                    newBuckets.add(bucket);
+                    continue;
+                }
+            }
+
+            for (Map.Entry<String, String> entry : bucketPathMap.entrySet()) {
+                String aggName = entry.getKey();
+                String bucketPath = entry.getValue();
+                Object propertyValue = resolveBucketValue(originalAgg, bucket, bucketPath);
+
+                if (propertyValue instanceof Number) {
+                    double doubleVal = ((Number) propertyValue).doubleValue();
+                    // NaN or infinite values indicate a missing value
+                    if (Double.isFinite(doubleVal)) {
+                        inputFields.put(aggName, doubleVal);
+                    }
+                } else if (propertyValue instanceof InternalNumericMetricsAggregation.SingleValue) {
+                    double doubleVal = ((InternalNumericMetricsAggregation.SingleValue) propertyValue).value();
+                    if (Double.isFinite(doubleVal)) {
+                        inputFields.put(aggName, doubleVal);
+                    }
+                } else if (propertyValue instanceof StringTerms.Bucket) {
+                    StringTerms.Bucket b = (StringTerms.Bucket) propertyValue;
+                    inputFields.put(aggName, b.getKeyAsString());
+                } else if (propertyValue instanceof String) {
+                    inputFields.put(aggName, propertyValue);
+                } else if (propertyValue != null) {
+                    // Doubles, String terms or null are valid, any other type is an error
+                    throw invalidAggTypeError(bucketPath, propertyValue);
+                }
+            }
+
+
+            InferenceResults inference;
+            try {
+                 inference = model.infer(inputFields, configUpdate);
+            } catch (Exception e) {
+                inference = new WarningInferenceResults(e.getMessage());
+            }
+
+            final List<InternalAggregation> aggs = StreamSupport.stream(bucket.getAggregations().spliterator(), false).map(
```

Comment:
I think we can do `bucket.getAggregations().asList().stream()...` instead of StreamSupport.  The older pipelines have that laying around because they predated stream on the iterator itself iirc.

Reply:
Guilty of copy-and-paste here.